osquery/tables/yara/yara.cpp

```diff
@@ -49,6 +50,27 @@ void doYARAScan(YR_RULES* rules,
   }
 }
 
+static bool shouldSkipFile(const std::string& path) {
+  struct stat sb;
+  if (0 != stat(path.c_str(), &sb)) {
+    return true; // failed to stat
+  }
+
+  // avoid special files /dev/x , /proc/x, FIFO's named-pipes, etc.
+  if ((sb.st_mode & S_IFMT) != S_IFREG) {
+    return true;
+  }
+
+  // avoid reading osquery DB LOCK files
+  if (path.find("osquery") != std::string::npos) {
+    if (boost::algorithm::ends_with(path,"LOCK")) {
```

Comment:
What problem does scanning LOCKs cause? I ask because I imagine there are lots of similar instances of file mutexes.

Reply:
specifically the osquery rocksdb lock.  This may have been preemptive... can't remember if there was an actual case on a specific platform, but wouldn't doubt it.